improve Edge documentation (#3677)
freqtrade/freqtrade: docs/edge.md

This page explains how to use Edge Positioning module in your bot in order to enter into a trade only if the trade has a reasonable win rate and risk reward ratio, and consequently adjust your position size and stoploss.

>   !!! Warning

Reviewer: Indenting these will cause rendering to break, as the tripple exclamation mark MUST be the first 3 letters to render correctly on readthedocs.

Applies to the below Note too, obviously.

    Edge positioning is not compatible with dynamic (volume-based) whitelist.

>   !!! Note
>  1. Edge does not consider anything other than *its own* buy/sell/stoploss signals. It ignores the stoploss, trailing stoploss, and ROI settings in the strategy configuration file.
> 
>  2. Therefore, it is important to understand that Edge can improve the performance of some trading strategies but *decrease* the performance of others.

Reviewer: I don't think this will render correctly. (`!!! Note` and the following, indented lines are rendered in special boxes in the documentation)

Once the indentation is broken (the empty line) - the box will end.

```suggestion
!!! Note
    Edge does not consider anything other than *its own* buy/sell/stoploss signals. It ignores the stoploss, trailing stoploss, and ROI settings in the strategy configuration file.
    Therefore, it is important to understand that Edge can improve the performance of some trading strategies but *decrease* the performance of others.
```

I'd not put it as numbered list, but simply as text - that should work out fine - considering it's rendered in a "note" box.
The text / wording itself is perfect though :+1:

| APPC/BTC  |      -0.02 |       0.44 |                2.28 |                   1.27 |         0.44 |                       25 |                       43 |
| NEBL/BTC  |      -0.03 |       0.63 |                1.29 |                   0.58 |         0.44 |                       19 |                       59 |

> Edge produced the above table by comparing ``calculate_since_number_of_days`` to ``minimum_expectancy`` to find ``min_trade_number``. Historical information based on the config file. The time frame Edge uses for its comparisons can be further limited by using the ``--timeframe`` switch.

Reviewer: I think what you mean with this is to limit the timerange, not the timeframe (timeframe = 5m - timerange would be "january-march" ...)
Unfortunately, the 2 words are very similar, so we need to be carefull to use them correctly.

```suggestion
Edge produced the above table by comparing `calculate_since_number_of_days` to `minimum_expectancy` to find `min_trade_number` historical information based on the config file. The timerange Edge uses for its comparisons can be further limited by using the `--timerange` switch.
```

Author: You are correct! My bad. Thanks for catching that.

> In live and dry-run modes, after the ``process_throttle_secs`` has passed, Edge will again process ``calculate_since_number_of_days`` against ``minimum_expectancy`` to find ``min_trade_number``. If no ``min_trade_number`` is found, the bot will return "whitelist empty". Depending on the trade strategy being deployed, "whitelist empty" may be return much of the time---or *all* of the time. The use of Edge may also cause trading to occur in bursts, though this is rare.

Reviewer: Maybe an additional sentence:

```suggestion
In live and dry-run modes, after the `process_throttle_secs` has passed, Edge will again process `calculate_since_number_of_days` against `minimum_expectancy` to find `min_trade_number`. If no `min_trade_number` is found, the bot will return "whitelist empty". Depending on the trade strategy being deployed, "whitelist empty" may be return much of the time - or *all* of the time. The use of Edge may also cause trading to occur in bursts, though this is rare.

If you encounter "whitelist empty" a lot, condsider tuning `calculate_since_number_of_days`, `minimum_expectancy`  and `min_trade_number` to align to the trading frequency of your strategy.
```

I'm not certain if the double-ticks (``) are fine or cause a problem on readthedocs (the rendered documentation page) - we're usually using single-ticks throughout the documentation (and that's also what markdown shows for inline code segments) - so i'd like to keep it that way.
